### PR TITLE
Fix emmet completions in scss

### PR DESCRIPTION
**Pull request by jeanp413**
Part of https://github.com/microsoft/vscode/issues/112675

### CODE REVIEW

File: src/emmetHelper.ts [hunk 1]
```diff
@@ -493,10 +493,6 @@ export function isAbbreviationValid(syntax: string, abbreviation: string): boole
 		return false;
 	}
 	if (isStyleSheet(syntax)) {
-		// Fix for https://github.com/Microsoft/vscode/issues/1623 in new emmet
```
Comment (rzhao271) [inline src/emmetHelper.ts:496]: Does this cause 1623 to show up again? Or is that case now being handled by the changes to `isExpandedTextNoise`?

File: src/emmetHelper.ts [hunk 2]
```diff
@@ -522,12 +518,19 @@ export function isAbbreviationValid(syntax: string, abbreviation: string): boole
...
 	// Unresolved css abbreviations get expanded to a blank property value
 	// Eg: abc -> abc: ; or abc:d -> abc: d; which is noise if it gets suggested for every word typed
 	if (isStyleSheet(syntax)) {
-		const after = (syntax === 'sass' || syntax === 'stylus') ? '' : ';';
-		return expandedText === `${abbreviation}: \${0}${after}` ||
+		const between = options['stylesheet.between'] ?? ': ';
+		const after = options['stylesheet.after'] ?? ';';
+
+		// Remove overlapping between `abbreviation` and `between`, if any
+		let endPrefixIndex = abbreviation.indexOf(between[0], abbreviation.length - between.length);
```
Comment (rzhao271) [inline src/emmetHelper.ts]: Can you give an example of when this code would apply?

File: src/emmetHelper.ts (hunk 1 above)
Comment (jeanp413) [reply src/emmetHelper.ts:496]: This check has been moved to https://github.com/microsoft/vscode/pull/118666/files

File: src/emmetHelper.ts (hunk 2 above)
Comment (jeanp413) [reply src/emmetHelper.ts]: Now that the above check is gone, typing `abc:` will generate an emmet completion and this change ensures that it's marked as noise and ignored.
The overlapping in this case would be `:` between `abc:` and `: `(colon and space)

Comment (rzhao271) [reply src/emmetHelper.ts]: While checking the `indexOf` there, I noticed that there's `lastIndexOf`. I wonder whether using that would be clearer?

Comment (jeanp413) [reply src/emmetHelper.ts]: yeah also thought about it but `lastIndexOf` will search from back to front and we only need to check the last `between.length` characters, using `lastIndexOf` will check the whole string if there's no match

we could  call `subtring` first and then `lastIndexOf` let me know if you prefer this

**Review by rzhao271 — APPROVED** [review]

**Pull request merged**